spec/models/conference_spec.rb

```diff
@@ -1639,4 +1639,26 @@
       expect(free_ticket.price_cents).to eq(0)
     end
   end
+
+  describe 'after_update' do
+    let(:conference) { create(:conference) }
+    let(:first_scheduled_event) { create(:event_scheduled, program: conference.program, hour: conference.start_date + conference.start_hour.hours) }
+    let(:second_scheduled_event) { create(:event_scheduled, program: conference.program, hour: conference.start_date + conference.end_hour.hours + 1.hour) }
+
+    it 'delete event schedules that are not in hour ranges, when conference start hour is updated' do
+      first_scheduled_event
+      conference.start_hour = conference.start_hour + 1.hour
+      conference.save
+      conference.run_callbacks :update
```

Comment:
> then there is already a conference variable at the top
But this is a already happened conference so I will have to change the start_date and end_date in the test so that I can successfully create my events. Therefore I didn't use it.